internal/provider/providers/porkbun/provider.go

```diff
@@ -138,8 +138,22 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		return ip, nil
 	}
 
-	for _, recordID := range recordIDs {
-		err = p.updateRecord(ctx, client, recordType, ipStr, recordID)
+	for _, record := range records {
+		matches := record.Name == (p.owner+"."+p.domain) ||
+			(p.owner == "@" && record.Name == p.domain)
+
+		if !matches {
+			// We fetch by domain, type, and owner. This should never happen
+			return netip.Addr{}, fmt.Errorf("received unexpected record for non matching owner: %w", errors.ErrUnknownResponse)
+		}
+
+		if record.Content == ipStr {
+			// Skip update. Porkbun already has the IP correctly set.
+			// Porkbun will respond with a 400 error if we attempt to update a record to a value that is already set.
```

Comment:
This is confusing, does that mean Porkbun returns 400 even if we try to update to another IP address than the one already set?

Reply:
if Porkbun has an DNS A record for `domain.tld` with an IP of `127.0.0.1`, and DDNS-Updater tries to update the A record for `domain.tld` to `127.0.0.1`, it will return a 400 error.